Pidgin/Parser.Number.cs

```diff
@@ -143,5 +145,54 @@ private static long GetLetterOrDigitValueLong(char c)
         }
         private static long GetUpperLetterOffsetLong(char c) => c - 'A';
         private static long GetLowerLetterOffsetLong(char c) => c - 'a';
+		
+		/// <summary>
+        /// A parser which parses a floating point number with an optional sign.
+        /// The resulting <c>double</c> is not checked for overflow.
+        /// </summary>
+        /// <returns>A parser which parses a floating point number with an optional sign</returns>
+        public static Parser<char, double> Float { get; } = RealNum();
+
+        /// <summary>
+        /// A parser which parses a floating point number with an optional sign.
+        /// The resulting <c>double</c> is not checked for overflow.
+        /// </summary>
+        /// <returns>A parser which parses a floating point number with an optional sign</returns>
+        public static Parser<char, double> RealNum()
+            => Map(
+                (sign, num) => sign.HasValue && sign.Value == '-' ? -num : num,
+                (Char('-').Or(Char('+'))).Optional(),
+                UnsignedReal()
+            ).Labelled($"real number");
+			
+		/// <summary>
+		/// A parser which parses a floating point number without a sign.
+		/// The resulting <c>double</c> is not checked for overflow.
+		/// </summary>
+		/// <returns>A parser which parses a floating point number without a sign.</returns>
+
+		public static Parser<char, double> UnsignedReal()
+			=> Map(
+				(integerPart, _, decimalPart) => integerPart.HasValue ? integerPart.Value + decimalPart : decimalPart,
+				UnsignedInt(10).Optional(),
+				Char('.'),
+				DigitCharDouble().ChainAtLeastOnceL(
```

Comment:
This basic structure does make sense, but I think if you're going to go to the effort of packing the number into a string and then parsing it out, you might as well just do the whole number that way. It's probably easier to write a custom `Parser` subclass at this point.

Reply:
My reasoning for splitting the process into two pieces was that we would not got through the conversion part (with the string allocation) unless we actually have a floating point number. It  eliminates any false positives.  I liked how you architected a beginning parser, then 'ChainAtLeastOnceL' as a callback.

Please provide guidance if this is a required change.

Reply:
I expect that `double.Parse` might run faster, but that needs testing of course. If you implement my suggested change to save the string allocation then that's no longer a concern.